lib/parsers.js

```diff
@@ -115,7 +115,7 @@ function getContextFromResponse (res, conf, isError) {
 
   if (isError) {
     context.headers_sent = res.headersSent
-    context.finished = res.finished
+    context.finished = res.finished || res.writableEnded
```

Comment:
@thesixpathguy 

There are some tests failing with the following message for node v12 and below

```
not ok 567 should be deeply equivalent
  ---
    operator: deepEqual
    expected: |-
      { status_code: 204, headers: { 'content-length': '4', string: 'foo', number: '42', array: [ 'foo', '42' ], password: '[REDACTED]' }, headers_sent: false, finished: false }
    actual: |-
      { status_code: 204, headers: { 'content-length': '4', string: 'foo', number: '42', array: [ 'foo', '42' ], password: '[REDACTED]' }, headers_sent: false }
    at: <anonymous> (/home/runner/work/apm-agent-nodejs/apm-agent-nodejs/test/agent.test.js:1793:11)
```

This is due to `writableEnded` not being defined on those versions therefore when response is not finished the expression becomes
```javascript
context.finished = false || undefined  // the result of the expression is undefined
```
then at serialisation of the span the process omits undefined properties to save bandwidth.

We can wrap the expression with a `Boolean` constructor to make sure we have the right type.
```suggestion
    context.finished = Boolean(res.finished || res.writableEnded)
```